docs/types/command.md

```diff
@@ -15,6 +15,110 @@ command('usage pattern', 'help page'):
     script
 ```
 
+## Options and arguments (usage pattern)
+
+`input.*` can also be used in expression filtered exec blocks (`|=`, see below). 
+
+### Arguments
+```
+command('cmd <required> [<optional>]'):
+  env:
+    REQUIRED: = input.argument('required')
+    OPTIONAL: = input.argument('optional')
+  exec: |
+    #!bash
+    echo "'${REQUIRED}'"
+    echo "'${OPTIONAL}'" 
+```
+Defines arguments. The command will not be recognized without required arguments.
+
+```
+command('cmd %'):
+  env:
+    ALL: = input.argument('%')
+  exec: |
+    #!bash
+    echo "'${ALL}'"
+```
+
+The special token `%` describes an argument that contains
+the rest of the input, it can only be the last character in the command.
+
+### Arguments with select values
+
+```
+command('cmd [on|off]'):
+  env:
+    OPTIONAL_ARG: = input.command(1)
+```
+or
+```
+command('cmd (on|off])'):
+  env:
+    REQUIRED_ARG: = input.command(1)
+```
+
+Defines arguments that can only have select values, these only work correctly
+as the last part in the command and must be read using `input.command(index)`, 0-based.
+
+### Boolean options
+```
+command('cmd [--option]'):
+  env:
+    VERBATIM: = input.option('option')
+    YES_OR_NO: = boolToString(input.option('option'))
```

Comment:
This is a [custom harness function](https://github.com/inviqa/harness-docker/blob/410656504a76adb59a03f493c8daaedc4036fdb6/harness/config/functions.yml#L280)

Reply:
You are right, I did think more of typical harness users here. Should I leave it in and give it a caveat comment or remove it?

Reply:
@kierenevans Like this maybe?